diff --git a/lib/asciidoctor/syntax_highlighter/pygments.rb b/lib/asciidoctor/syntax_highlighter/pygments.rb
index 191acb31..08aa63e9 100644
--- a/lib/asciidoctor/syntax_highlighter/pygments.rb
+++ b/lib/asciidoctor/syntax_highlighter/pygments.rb
@@ -121,7 +121,7 @@ class SyntaxHighlighter::PygmentsAdapter < SyntaxHighlighter::Base
     end
 
     DEFAULT_STYLE = 'default'
-    BASE_SELECTOR = 'pre.pygments'
+    BASE_SELECTOR = '.pygments'
     TOKEN_CLASS_PREFIX = 'tok-'
 
     BaseStyleRx = /^#{BASE_SELECTOR.gsub '.', '\\.'} +\{([^}]+?)\}/
